packages/terra-signature/src/Signature.jsx

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'terra-base/lib/baseStyles';
+import styles from './Signature.scss';
+
+const LineWidths = {
+  EXTRA_FINE: 1,
+  FINE: 2,
+  MEDIUM: 4,
+  HEAVY: 6,
+};
+const propTypes = {
+  /**
+   * The line width to use when drawing the signature on the canvas.
+   * One of LineWidths.EXTRA_FINE, LineWidths.FINE, LineWidths.MEDIUM, LineWidths.HEAVY.
+   */
+  lineWidth: PropTypes.oneOf([LineWidths.EXTRA_FINE, LineWidths.FINE, LineWidths.MEDIUM, LineWidths.HEAVY]),
+  /**
+  * Line segments that define signature.
+  */
+  lineSegments: PropTypes.array,
+  /**
+   * A callback function to execute when a line segment is drawn. The first parameter is the event, the
+   * second parameter is all the line segments, and the last parameter is the most recent line segment drawn.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  lineWidth: LineWidths.FINE,
+  lineSegments: [],
+  onChange: undefined,
+};
+
+class Signature extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = { lineSegments: props.lineSegments, lineWidth: props.lineWidth };
+
+    this.mouseDown = this.mouseDown.bind(this);
+    this.mouseUp = this.mouseUp.bind(this);
+    this.mouseXY = this.mouseXY.bind(this);
+    this.addLine = this.addLine.bind(this);
+    this.draw = this.draw.bind(this);
+    this.drawSignature = this.drawSignature.bind(this);
+    this.clearSignature = this.clearSignature.bind(this);
+    this.updateDimensions = this.updateDimensions.bind(this);
+  }
+
+  componentDidMount() {
+    if ('ontouchstart' in document.documentElement) {
+      this.canvas.addEventListener('touchstart', this.mouseDown, false);
+      this.canvas.addEventListener('touchmove', this.mouseXY, true);
+      this.canvas.addEventListener('touchend', this.mouseUp, false);
+      document.body.addEventListener('touchcancel', this.mouseUp, false);
+    } else {
+      this.canvas.addEventListener('mousedown', this.mouseDown);
+      this.canvas.addEventListener('mousemove', this.mouseXY);
+      document.body.addEventListener('mouseup', this.mouseUp);
+    }
+
+    const context = this.canvas.getContext('2d');
```

Comment:
Update to `this.context = this.canvase.getContext('2d')` and remove `const context = this.canvas.getContext('2d');` from draw, drawSignature and clearSignature

Reply:
Please address this comment.

Reply:
Checked with Emily regarding this. We cannot have an instance variable for the getContext method due to the fact that on every redraw that happens the canvas reference will change and so the getContext reference needs to be updated as well. We tried another approach by pointing the canvas ref to a function that sets both values, but it didn't worked either. In the end it was decided to leave those const generated when needed.